src/features/courses/course-route.ts

```diff
@@ -205,17 +202,51 @@ router.get('/questions',
 
         let topic: CourseTopicContent | null = null;
         if(!_.isNil(req.query.courseTopicContentId)) {
-            topic = await courseController.getTopicById(req.query.courseTopicContentId, userId);
+            topic = await courseController.getTopicById({
+                id: req.query.courseTopicContentId,
+                userId,
+            });
             const overrideStartDate = topic.studentTopicOverride?.[0]?.startDate;
             const startDate = overrideStartDate ?? topic.startDate;
 
+            const user = await req.session.getUser();
             if (moment().isBefore(startDate)) {
-                const user = await req.session.getUser();
                 if (user.roleId === Role.STUDENT) {
                     next(Boom.badRequest(`The topic "${topic.name}" has not started yet.`));
                     return;
                 }
             }
+
+            if (topic?.topicTypeId === 2 && !_.isNil(userId)) {
+                if (_.isNil(topic.topicAssessmentInfo)){
+                    next(Boom.badRequest('Topic is an assessment, but does not have corresponding assessment info. This should never happen.'));
+                    return;
+                }
+
+                // get version info - descending by startTime unless specific id is included in query
+                let versions: StudentTopicAssessmentInfo[] | undefined;
+                if (_.isNil(req.query.studentTopicAssessmentInfoId)) {
+                    versions = _.orderBy(topic.topicAssessmentInfo.studentTopicAssessmentInfo, ['startTime'],'desc');
+                    if (_.isNil(versions) || versions.length === 0) {
+                        next(httpResponse.Ok('You have not started any versions of this assessment.', {questions: null, topic}));
+                        return;
+                    }
+                } else {
+                    const version = await courseController.getStudentTopicAssessmentInfoById(req.query.studentTopicAssessmentInfoId);
+                    versions = [version];
+                }
+
+                if (
+                    topic.topicAssessmentInfo.hideProblemsAfterFinish && (
+                        moment().isAfter(moment(versions[0].endTime)) ||
+                        topic.topicAssessmentInfo.maxGradedAttemptsPerVersion <= versions[0].numAttempts // TODO: replace with versions[0].isFinished
+                    ) &&
+                    user.roleId === Role.STUDENT
+                ) {
+                    next(httpResponse.Ok('You have finished this version of the assessment and you are blocked from seeing the problems.', {questions: null, topic}));
```

Comment:
Better verbiage might be "Problems on this exam many not be viewed until the exam has finished for all students ${endDate}"

Reply:
Probably true, although I don't know if we use this message from the server, most of the time we don't
That being said it is still good to keep it clean

Reply:
I want to be able to use these messages in the modal that is displayed to students on the front end.